### PR TITLE
Document `torch.cuda.profiler.start`

### PR DESCRIPTION
**Pull request by anandptl84**
document https://github.com/pytorch/pytorch/issues/127917 start function of cuda/ profiler.py

Fixes 127917

cc @svekars @brycebortree @ptrblck @msaroufim @robieta @chaekit @aaronenyeshi @guotuofeng @guyang3532 @dzhulgakov @davidberard98 @briancoutinho @sraikund16 @sanrise

### CODE REVIEW

**Review by aaronenyeshi — APPROVED** [review]
Could we also document the other apis?

File: torch/cuda/profiler.py [hunk 1]
```diff
@@ -45,6 +45,13 @@ def init(output_file, flags=None, output_mode="key_value"):
 
 
 def start():
+    """
+    Starts Cuda profile data collection
+
+    Example:
+        >>> profiler = Profiler();
+        >>> profiler.start()
```
Comment (davidberard98) [inline torch/cuda/profiler.py:53]: I think the usage would be more like `torch.cuda.profiler.start()`, I don't think there's a `Profiler()` object that would have to be constructed first

**Review by davidberard98 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**